### PR TITLE
Update homePagePreferenceSpec test describe() and it() statements

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Had this as part of some local changes I was going to push to #2736 in an attempt to figure out the `localStorage` `spyOn` quirks in Firefox.  I'm good to punt on that, seems a big hassle, but prob the updated test language is worth salvaging.

### CODE REVIEW

File: test/spec/services/homePagePreferenceSpec.js [hunk 1]
```diff
@@ -29,39 +29,57 @@ describe("HomePagePreferenceService", function(){
...
+
+         expect(homePagePreference).toEqual("/project/test-pro/overview?isHomePage=true");
+       });
+     });
+
+     describe('when given a config object with an invalid `type` option', function() {
+       // in the UI, this will also trigger a notification that the project is
+       // invalid.  This additional functionality should be verified either with
+       // a unit test or an e2e test against the appropriate controller.
+       it("should not set the home page", function(){
```
Comment (spadgett) [inline test/spec/services/homePagePreferenceSpec.js]: This is really testing that `notificationInvalidProjectHomePage` clears the value. The description is better, but not quite accurate.

Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: True, this should be in a separate `describe('#notifyInvalidProjectHomePage')`.  I'll update.

Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: Updated.

File: test/spec/services/homePagePreferenceSpec.js [hunk 2]
```diff
@@ -29,39 +29,59 @@ describe("HomePagePreferenceService", function(){
...
+         expect(homePagePreference).toEqual("/project/test-pro/overview?isHomePage=true");
+       });
+     });
+  });
+
+  describe('#notifyInvalidProjectHomePage', function() {
+    describe('when given a config object with an invalid `type` option', function() {
+      // in the UI, this will also trigger a notification that the project is
+      // invalid.  This additional functionality should be verified either with
+      // a unit test or an e2e test against the appropriate controller.
```
Comment (yapei) [inline test/spec/services/homePagePreferenceSpec.js]: Yeah, that's what I planned also, to cover the notifications checking in SetHomePageModalController unit/e2e testing, but I didn't add them in statement, as well as inappropriate language/description which cause confusion, sorry for this.

**Pull request merged**